source/lib-intro.tex

```diff
@@ -946,6 +946,9 @@
 Operators in expressions\iref{over.match.oper} are not so constrained;
 see \ref{global.functions}.
 \end{note}
+The meaning of the unqualified name \tcode{swap} is established
+in an overload resolution context
+for swappable values\iref{swappable.requirements}.
 
```

Comment:
On line 943, there's an example for `is_array_v`, which gives help where none is needed (type and variable names never change their lookup results based on qualification anyway); but there is no example given for line 937, which is where help _is_ needed (because it is the function lookup rules that are complicated already, and which are being changed by this wording). I suggest adding a new note, e.g.
```
\begin{note}
The reference to \tcode{swap_ranges} in the specification of \tcode{std::swap}\iref{utility.swap}
refers to \tcode{::std::swap_ranges}.
\end{note}
```

Reply:
Given that `swap` is the special case called out in this wording, using the definition of `std::swap` for an example where the special case doesn't apply can only lead to confusion.

Reply:
And I don't think this is needed anyway. The rule is consistent: unless specified otherwise, we do qualified lookup. Function or not, you do qualified lookup. So the extra complication that applies to lookup for functions (ADL) isn't relevant, we're explicitly not doing that.

Reply:
> can only lead to confusion...

...or clarification. ;)

The rule from the implementor's point of view is "Variable and type names are treated as written (e.g. when the paper standard says `is_same_v`, the implementor should write `is_same_v` in the code). Function names get special treatment (e.g. when the paper standard says `swap_ranges`, the implementor should write `std::swap_ranges` in the code)." The extra complication exists _only_ for function names, and yet that's the one case where no clarifying example is provided.

Reply:
@jensmaurer wrote:
> [Actually] the implementer needs to use [`::std::swap_ranges`] to disambiguate properly.

True. To rephrase:

> The rule from the implementor's point of view is "Variable and type names are treated as written. Function names get special treatment." The extra complication exists _only_ for function names, and yet that's the one case where no clarifying example is provided.

Proving that the special treatment of functions is _even more special_ than previously indicated, merely strengthens the force of the top-level statement.